src/commonMain/kotlin/fr/acinq/lightning/channel/InteractiveTx.kt

```diff
@@ -108,30 +105,35 @@ sealed class InteractiveTxInput {
     sealed class Local : InteractiveTxInput(), Outgoing {
         abstract val previousTx: Transaction
         abstract val previousTxOutput: Long
-        override val txOut: TxOut
-            get() = previousTx.txOut[previousTxOutput.toInt()]
+        override val outPoint: OutPoint get() = OutPoint(previousTx, previousTxOutput)
+        override val txOut: TxOut get() = previousTx.txOut[previousTxOutput.toInt()]
     }
 
     /** A local-only input that funds the interactive transaction. */
-    data class LocalOnly(override val serialId: Long, override val previousTx: Transaction, override val previousTxOutput: Long, override val sequence: UInt) : Local() {
-        override val outPoint: OutPoint = OutPoint(previousTx, previousTxOutput)
-    }
+    data class LocalOnly(override val serialId: Long, override val previousTx: Transaction, override val previousTxOutput: Long, override val sequence: UInt) : Local()
 
-    /** A local input that funds the interactive transaction, coming from a 2-of-2 swap-in transaction. */
-    data class LocalLegacySwapIn(override val serialId: Long, override val previousTx: Transaction, override val previousTxOutput: Long, override val sequence: UInt, val userKey: PublicKey, val serverKey: PublicKey, val refundDelay: Int) : Local() {
-        override val outPoint: OutPoint = OutPoint(previousTx, previousTxOutput)
-        override val txOut: TxOut = previousTx.txOut[previousTxOutput.toInt()]
-    }
+    /** A local input that funds the interactive transaction, coming from a 2-of-2 legacy (pay2wsh) swap-in transaction. */
+    data class LocalLegacySwapIn(
+        override val serialId: Long,
+        override val previousTx: Transaction,
+        override val previousTxOutput: Long,
+        override val sequence: UInt,
+        val userKey: PublicKey,
+        val serverKey: PublicKey,
+        val refundDelay: Int
+    ) : Local()
 
+    /** A local input that funds the interactive transaction, coming from a 2-of-2 swap-in transaction. */
     data class LocalSwapIn(
         override val serialId: Long,
         override val previousTx: Transaction,
         override val previousTxOutput: Long,
         override val sequence: UInt,
-        val swapInParams: TxAddInputTlv.SwapInParams
-    ) : Local() {
-        override val outPoint: OutPoint = OutPoint(previousTx, previousTxOutput)
-    }
+        val userKey: PublicKey,
```

Comment:
I wasn't a big fan of this, because it was coupling two different use-cases, I didn't like that the `InteractiveTx` types had to depend on spec TLVs. The spec TLVs may contain data that isn't useful here (we saw that when we attempted to get rid of the `prevtx`: the `SwapInParams` tlvs contained outpoint and txOut information that ended up being redundant in the `InteractiveTx` types). It's a bit more verbose to explicitly spell out every field, but it guarantees we have exactly the data we need.

Reply:
Maybe but the same could be said of methods in `SwapInProtocol` that reference TxSignature TLVs

Reply:
That's true, and I think it would be cleaner if `SwapInProtocol` didn't use the `TxSignatures` tlvs! Do you want me to change that?

Reply:
Done in https://github.com/ACINQ/lightning-kmp/pull/592/commits/47df69e3f1dd70b51a25aac51f22a2eb415d211e

Reply:
I'd rather keep referencing `SwapInParams` :) but i's 55/45. But yes if you want to avoid depending on TLV types you should try and remove `TxSignatures` from `SwapInProtocol`.  To me it feels like in both cases using these TLVS is the simplest and most direct choice.